src/dialogsettings.cpp

```diff
@@ -525,3 +587,24 @@ bool DialogSettings::isMorkParserSelected() const
 {
     return boxParserSelection->currentIndex() == 1;
 }
+
+QStringList DialogSettings::searchThunderbird() const {
+    QStringList defaultCommand = Utils::getDefaultThunderbirdCommand();
+    if (defaultCommand.count() == 1 && !QFileInfo(defaultCommand[0]).isExecutable()) {
+        return defaultCommand;
+    }
+    QString thunderbirdPath = QStandardPaths::findExecutable("thunderbird");
+    if (!thunderbirdPath.isEmpty()) {
+        return {thunderbirdPath};
+    }
+    if (QApplication::applicationDirPath().contains("flatpak")) { // TODO: Is this the best way to detect if we are installed with flatpak and does it even work?
+        // TODO: Is this really the best way to check for other apps?
```

Comment:
@ThanosApostolou This is where we could detect that we are installed as a flatpak app and detect the Thunderbird installations. Maybe we can detect them [by their desktop files](https://github.com/flatpak/flatpak/issues/1286#issuecomment-354554684), if we can see the files in our sandbox?
Unfortunately, I don't have the setup to test this, so I would leave this part for you, if that is ok for you. If the tests would become too complicated, we could instead just return the default `{"/usr/bin/flatpak-spawn", "--host", "flatpak", "run", "org.mozilla.Thunderbird"`} here.

Reply:
Guys, I suggest we simply prepackage the configuration file for flatpaks and not code complex heuristics for executable lookups for this purpose. My understanding is that there would be two flatpacs - one using built-in Thunderbird, and one without Thunderbird. They would come with a different configuration file and start different clients.

I'd rather avoid hard-coding a list of possible thunderbird executables and trying to look them up anywhere in the system. Not only this is wrong from the programming standpoint, but you also would have no guarantee that Thunderbird you've found is the one user actually wants to use.

Reply:
Is distributing two different versions of Birdtray really the right answer? Or three versions, if you want to support the snap version? I'm not suggesting hardcoding a list of possible locations, but rather search for their app id once we know that we have to call them via the flatpak command.
I agree that we might pick up a different Thunderbird than the user is using (but why have two differen versions installed anyway?), but this component is only ment to autodetect *any* Thunderbird installation.

But if you don't like the added complexity, we can always just return the standard flatpak command.

Reply:
Distributing the same birdtray executable packaged with two/three different configuration files is a possible answer. Another is wrapping the setup into an installer which would ask the user which Thunderbird they'd like to use, and create the right configuration for Birdtray. This is however something which should be done by the person creating the Flatpak, and not by us.

Ideally I prefer to have no code at all inside the Birdtray which does anything flatpak- or snap-specific. Those are supposed to be application wrappers; it is their job to provide the functionality relevant for wrapping the application.

Reply:
> ... it is their job to provide the functionality relevant for wrapping the application.

You do have a point there. Ok, let's keep the searching for Thunderbird as simple as possible.